composer/loggers/progress_bar_logger.py

```diff
@@ -305,11 +313,19 @@ def init(self, state: State, logger: Logger) -> None:
             )
         self.state = state
 
+    def fit_start(self, state: State, logger: Logger) -> None:
+        if not self.hparams_already_logged_to_console:
+            self.hparams_already_logged_to_console = True
+            self._log_hparams_to_console()
+
     def epoch_start(self, state: State, logger: Logger) -> None:
         if self.show_pbar and not self.train_pbar:
             self.train_pbar = self._build_pbar(state=state, is_train=True)
 
     def eval_start(self, state: State, logger: Logger) -> None:
+        if not self.hparams_already_logged_to_console:
+            self.hparams_already_logged_to_console = True
+            self._log_hparams_to_console()
```

Comment:
Should this be on rank 0 only?

Also, can we do the same for `predict_start`

Reply:
it is rank_zero only. See `_log_hparams_to_console` function.

Reply:
> > Can you explain why we need to do this lazily at start of fit/eval/predict instead of end of init?
> 
> Because we want them all printed out in a block all at once and the hyperparameters come in at different times (i.e. `log_hyperparameters` is called multiple times). Also, there is no INIT_END event (there should be!)

What about `Event.AFTER_LOAD`? https://github.com/mosaicml/composer/pull/1652

Reply:
ok added `predict_start` support

Reply:
Can we move to `Event.AFTER_LOAD` instead? or do you not like that. Seems cleaner to actually give the hyperparameters after calling init and less duplicated code

Reply:
> What about `Event.AFTER_LOAD`? #1652

Wow timely PR merge. The issue is that people may do this:
```
args = parser.parse_args()
trainer = Trainer(model=model, ... )
trainer.logger.log_hyperparameters(**vars(args))
trainer.fit()
```
So logging of the hyperparameters will happen after `__init__` returns

Reply:
also, see https://github.com/mosaicml/composer/blob/dev/examples/run_composer_trainer.py#L47

Reply:
Hm... ok. Feels a little hacky, but agree with you.

Reply:
yeah well very soon we will have auto-hparam logging, so we won't have that pattern.